### ISSUE TITLE
Long messages are cut off.

### ISSUE THREAD

**Issue by ThePhilgrim** [Akuli/mantaray#172]
![Skärmavbild 2022-01-11 kl  18 38 58](https://user-images.githubusercontent.com/76887896/148993495-38534312-9bac-4026-bb23-17c88c6c9302.png)

**Comment by Akuli** [comment]
Other IRC clients send multiple messages if the message is too long. We should figure out:
- What is the maximum length for a message on libera?
- Is it server-specific? Does a server tell the client about it when a client connects?
- Does the maximum length include `PRIVMSG #mantatest` in the beginning and `\r\n` at the end, or only the number of bytes in the message itself?

**Comment by Akuli** [comment]
I read ircdocs and experimented with libera a bit. The maximum length is determined by your user mask, as follows: if you're usermask is `foo!bar@baz.com`, then `b":foo!bar@baz.com PRIVMSG #channel :message\r\n"` must fit in 512 bytes. This string is what the server ends up sending to other clients. So the maximum length for message is:

```
512 - 14 - len(user_mask) - len(channel_name)
```

where `14` is the length of the above line without the user mask and channel name.

This is difficult to calculate, because finding your own user mask seems to be more difficult than I expected:
- I can't simply do `nick!user@host`, because `user` may get truncated. It may also get `~` in front, or if you have identd, something completely different depending on what identd says. For example, my user mask while testing this was `akuli-testing!~akuli-tes@12-345-678-901.bb.dnainternet.fi` (where the numbers were actually my IP, not sure if I want to write that to github). If the server provided `USERLEN` in `RPL_ISUPPORT`, this would be easy, but libera doesn't send `USERLEN`.
- When I connect to libera and join a chanel, I see my own user mask only when I have finished joining a channel. But that won't help if I want to send private message to someone without joining any channels.
- If I change my nick after the initial registration, the server does send me the mask. But the mask includes my old nick, and I have to substitute in the new nick myself.

tl;dr this is tricky :)

**Comment by Moosems** [comment]
Wrap by words rather than characters.
